pulsar-client-1x-base/pulsar-client-1x/src/main/java/org/apache/pulsar/client/api/ProducerConfiguration.java

```diff
@@ -101,7 +101,7 @@ public long getSendTimeoutMs() {
      *            the time unit of the {@code sendTimeout}
      */
     public ProducerConfiguration setSendTimeout(int sendTimeout, TimeUnit unit) {
-        checkArgument(sendTimeout >= 0);
+        checkArgument(sendTimeout >= 0, "sendTimeout needs to be >= 0");
```

Comment:
it seems that you moved the validations to ProducerConfigurationData for other methods. doesn't it make sense to move this method as well?

Reply:
Thanks for review. The following functions require `TimeUnit` as well (apart from the others). I agree and addressed at last patch.
```
sendTimeout(int sendTimeout, @NonNull TimeUnit unit)
batchingMaxPublishDelay(long batchDelay, @NonNull TimeUnit timeUnit)
```